Tests/Unit/Twig/DescriptionExtensionTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2016 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\ResourceBundle\Tests\Unit\Twig;
+
+use Symfony\Cmf\Component\Resource\Description\DescriptionFactory;
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+use Symfony\Cmf\Component\Resource\Description\Description;
+use Symfony\Cmf\Bundle\ResourceBundle\Twig\DescriptionExtension;
+
+class DescriptionExtensionTest extends \PHPUnit_Framework_TestCase
+{
+    private $descriptionFactory;
+    private $extension;
+    private $resource;
+    private $description;
+
+    public function setUp()
+    {
+        $this->descriptionFactory = $this->prophesize(DescriptionFactory::class);
+        $this->extension = new DescriptionExtension(
+            $this->descriptionFactory->reveal()
+        );
+        $this->resource = $this->prophesize(CmfResource::class);
+        $this->description = $this->prophesize(Description::class);
+    }
+
+    /**
+     * It should provide the cmf_resource_description function.
+     */
+    public function testProvideResourceDescription()
+    {
+        $this->descriptionFactory->getPayloadDescriptionFor($this->resource->reveal())->willReturn($this->description->reveal());
+        $function = $this->getFunction('cmf_resource_description');
+        $this->assertNotNull($function);
+        $callable = $function->getCallable();
+        $description = $callable($this->resource->reveal());
+
+        $this->assertInstanceOf(Description::class, $description);
+        $this->assertSame($this->description->reveal(), $description);
```

Comment:
Well, I don't see any need to make it a functional test, we can safely
test only the Extension class here (as we are doing).

The only issue I see is that the test is testing both the `getFunctions`
method and the function that it returns.

What if we split it into two tests?

On Sun, Jun 05, 2016 at 04:59:57AM -0700, Wouter J wrote:
>    In [1]Tests/Unit/Twig/DescriptionExtensionTest.php:
> 
>  > +        $this->description = $this->prophesize(Description::class);
>  > +    }
>  > +
>  > +    /**
>  > +     * It should provide the cmf_resource_description function.
>  > +     */
>  > +    public function testProvideResourceDescription()
>  > +    {
>  > +        $this->descriptionFactory->getPayloadDescriptionFor($this->resource->reveal())->willReturn($this->description->reveal());
>  > +        $function = $this->getFunction('cmf_resource_description');
>  > +        $this->assertNotNull($function);
>  > +        $callable = $function->getCallable();
>  > +        $description = $callable($this->resource->reveal());
>  > +
>  > +        $this->assertInstanceOf(Description::class, $description);
>  > +        $this->assertSame($this->description->reveal(), $description);
> 
>    I don't like this test very much. What about converting it to a functional
>    test? (unit testing it doesn't make much sense, as the unit doesn't
>    contain logic)
> 
>  // ...
>  class DescriptionExtensionTest extends \PHPUnit_Framework_TestCase
>  {
>      private $twig;
>      private $descriptionFactory;
> 
>      protected function setUp()
>      {
>          $this->descriptionFactory = $this->prophesize(DescriptionFactory::class);
> 
>          $this->twig = new \Twig_Environment(new \Twig_Loader_Array([]));
>          $this->twig->addExtension(new DescriptionExtension($this->descriptionFactory->reveal()));
>      }
> 
>      public function testProvidesResourceDescription()
>      {
>          $resource = $this->prophesize(CmfResource::class)->reveal();
> 
>          $description = $this->prophesize(Description::class);
>          $description->get('title')->willReturn('Hello world!');
> 
>          $this->descriptionFactory->getPayloadDescriptionFor($resource)->willReturn($description->reveal());
> 
>          $template = $this->twig->createTemplate('{{ cmf_resource_description(resource).get("title") }}');
>          $result = $template->render([
>              'resource' => $resource,
>          ]);
> 
>          $this->assertEquals('Hello world!', $result);
>      }
>  }
> 
>    —
>    You are receiving this because you modified the open/close state.
>    Reply to this email directly, [2]view it on GitHub, or [3]mute the thread.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/symfony-cmf/resource-bundle/pull/29#discussion_r65816346
>    2. https://github.com/symfony-cmf/resource-bundle/pull/29/files/8574363dadb1107deddf9e15540ae84dadf67332#r65816346
>    3. https://github.com/notifications/unsubscribe/AAgZcXN50XTgJiIAwK8BLdCuka6EZQBdks5qIro9gaJpZM4IpFET
>    4. https://github.com/symfony-cmf/resource-bundle/pull/29/files/8574363dadb1107deddf9e15540ae84dadf67332#r65816346